js/src/components/PostSettingsModal.js

```diff
@@ -0,0 +1,83 @@
+import { Button } from "@yoast/components/src/button/Button";
+import styled from "styled-components";
+import { __, sprintf } from "@wordpress/i18n";
+import { useState, useCallback } from "@wordpress/element";
+import { Modal } from "@wordpress/components";
+
+/**
+ * Container for the button.
+ *
+ * Sets the flex-direction to column to make the button full-width.
+ */
+const Container = styled.div`
+	padding: 16px;
+	display: flex;
+	flex-direction: column;
+	border-bottom: 1px solid rgba( 0,0,0,0.2 );
+`;
+
+const YoastIcon = styled.span`
+	background-color: var(--yoast-color-primary);
+	mask-image: var(--yoast-svg-icon-yoast);
+	-webkit-mask-image: var(--yoast-svg-icon-yoast);
+	mask-size: 100% 100%;
+	-webkit-mask-size: 100% 100%;
+	display: inline-block;
+	width: 20px;
+	height: 20px;
+	margin-right: 8px;
+`;
+
+const YoastModal = styled( Modal )`
+	width: 880px;
+	height: 100%;
+	max-width: calc(100% - 48px - 48px);
+	max-height: calc(100% - 48px - 48px);
+
+	@media only screen and (max-width: 600px){
+		max-height: 100%;
+		max-width: 100%;
+	}
+`;
+
+/**
+ * Returns a button in a div that can be used to open the modal.
+ *
+ * Warning: contains styling that is specific for the Sidebar.
+ *
+ * @returns {*} A button wrapped in a div.
+ */
+const PostSettingsModal = () => {
+	const [ isOpen, changeIsOpen ] = useState( false );
+
+	const closeModal = useCallback( () => changeIsOpen( false ), [] );
+	const openModal = useCallback( () => changeIsOpen( true ), [] );
+
+	return (
+		<Container
+			className="yoast"
+		>
+			{ isOpen && (
+				<YoastModal
+					title="Yoast SEO post settings"
+					icon={ <YoastIcon /> }
+					onRequestClose={ closeModal }
+					className="yoast yoast-gutenberg-modal"
+				>
+					Here comes the content.
+				</YoastModal>
```

Comment:
Perhaps this can be a reusable modal that already sets a basic `className` and `YoastIcon`, to DRY things a little. If you isolate that to a single commit, we could even cherrypick it to `trunk` and then special projects could use it too.